vignettes/data-preparation.Rmd

```diff
@@ -0,0 +1,399 @@
+
+# Create dfAnalyticsInput for participant-level drill-down
+dfAnalyticsInput <- purrr::map(lAnalyzed, "Analysis_Input") %>%
+  dplyr::bind_rows(.id = "MetricID") %>%
+  dplyr::as_tibble() %>%
+  dplyr::mutate(SnapshotDate = Sys.Date())
+```
+
+The reporting data model is documented in detail at [Reporting Data](https://gilead-biostats.github.io/gsm.core/articles/DataModel.html#reporting-data).
+
+# App Deployment
+
+Now that our data are ready let's look at how to deploy an app.
+
+## Creating the fnFetchData Function
+
+The `fnFetchData` function is essential for the {gsm.app} to provide drill-down functionality for domain-specific data. This function tells the app how to fetch domain data and should return filtered data based on the parameters passed from the app interface.
+
+```{r create-fnfetchdata, eval = FALSE}
+fnFetchData <- function(strDomainID, strGroupLevel = NULL, strGroupID = NULL, 
+                       strSubjectID = NULL, dSnapshotDate = NULL) {
+  
+  # Use the sample function as a template
+  # In your implementation, replace this with code that fetches from your data sources
+  gsm.app::sample_fnFetchData(
+    strDomainID = strDomainID,
+    strGroupLevel = strGroupLevel, 
+    strGroupID = strGroupID,
+    strSubjectID = strSubjectID,
+    dSnapshotDate = dSnapshotDate
+  )
+}
+```
+
+
+## Running the App
+
+Once you have prepared all the required data structures and the `fnFetchData` function, you can deploy the {gsm.app}.
+
+For development and testing, run the app locally: run_gsm_app() creates a Shiny app to explore a set of clinical trial data. The app facilitates exploration of the data by allowing the user to click to diver deeper into aspects of the data.
+
+
+```{r local-deployment, eval = FALSE}
+# Create the app data structure
+run_gsm_app(
+  dfAnalyticsInput = dfAnalyticsInput,
+  dfBounds = dfBounds,
+  dfGroups = dfGroups,
+  dfMetrics = dfMetrics,
+  dfResults = dfResults,
+  fnFetchData = fnFetchData,
+  fnCountData = ConstructDataCounter(fnFetchData),
+  chrDomains = c(AE = "Adverse Events", DATACHG = "Data Changes", DATAENT = "Data Entry",
+    ENROLL = "Enrollment", LB = "Lab", PD = "Protocol Deviations", QUERY = "Queries",
+    STUDCOMP = "Study Completion", SUBJ = "Subject Metadata", SDRGCOMP =
+    "Treatment Completion"),
+  lPlugins = NULL,
+  strTitle = ExtractAppTitle(dfGroups),
+  strFavicon = "angles-up",
+  strFaviconColor = "#FF5859",
+  tagListExtra = NULL,
+  fnServer = NULL
+)
+```
+
+# Troubleshooting
+
+## Common Issues
+
+1.  **Data Validation Errors**: Ensure all required columns are present and correctly named
+2.  **Missing MetricID Links**: Verify MetricID values are consistent across all data frames
+3.  **Date Format Issues**: Ensure all date columns are properly formatted as Date objects
+4.  **Memory Issues**: For large datasets, consider data filtering and optimization
+5.  **Fetch Function Errors**: Test your `fnFetchData` function with various parameter combinations
```

Comment:
Mention that errors are passed on to the user through the UI. I'd put this specifically into the fnFetchData() section and leave it off as a common issue. I'm... actually not certain that any of these ARE common issues. At most we have "Things we've noticed in our pilots" so far...